packages/medusa/src/api/admin/views/[entity]/columns/route.ts

```diff
@@ -0,0 +1,352 @@
+import { HttpTypes } from "@medusajs/framework/types"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "@medusajs/framework/http"
+import { MedusaModule } from "@medusajs/framework/modules-sdk"
+import {
+  GraphQLObjectType,
+  makeExecutableSchema,
+  mergeTypeDefs,
+  graphqlSchemaToFields,
+  extractRelationsFromGQL,
+  cleanGraphQLSchema,
+  print,
+  MedusaError,
+} from "@medusajs/framework/utils"
+import {
+  formatFieldName,
+  getColumnCategory,
+  getTypeInfoFromGraphQLType,
+  getUnderlyingType,
+  isArrayField,
+  isSingleRelationship,
+  shouldExcludeField,
+} from "./helpers"
+import { ENTITY_MAPPINGS } from "./entity-mappings"
+
+export const DEFAULT_COLUMN_ORDERS: Record<string, number> = {
+  display_id: 100,
+  created_at: 200,
+  customer_display: 300,
+  "sales_channel.name": 400,
+  fulfillment_status: 500,
+  payment_status: 600,
+  total: 700,
+  country: 800,
+}
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse<HttpTypes.AdminOrderColumnsResponse>
+) => {
+  const entity = req.params.entity
+
+  const entityMapping = ENTITY_MAPPINGS[entity as keyof typeof ENTITY_MAPPINGS]
+  if (!entityMapping) {
+    return res.status(400).json({
+      message: `Unsupported entity: ${entity}`,
+      type: "invalid_data",
+    } as any)
+  }
+
+  try {
```

Comment:
Can we have this schema introspection extracted to a util?

Reply:
yeah - I definitely think we should; that said I suspect this will evolve a little more in the near future so might make sense to wait with extracting it right now.